psiminer-core/src/main/kotlin/psi/nodeProperties/Token.kt

```diff
@@ -6,6 +6,7 @@ import com.intellij.psi.PsiElement
 import com.intellij.psi.util.PsiTreeUtil
 
 const val EMPTY_TOKEN = "EMPTY"
+const val EMPTY_TOKEN_OLD = "<EMPTY>"
```

Comment:
I really don't like this old token logic. It complicates psiminer logic and is required only for your one model (new models may be trained with a new format). Therefore, I may only suggest you keep the empty token as it is and add additional preprocessing on your side: replace all EMPTY with <EMPTY>.

Reply:
Apart from that, let's change the valid code token `EMPTY` to `<E>`. It's better both from memory usage and later detect sides.